internal/ocm/client.go

```diff
@@ -0,0 +1,175 @@
+limitations under the License.
+*/
+
+package ocm
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/rosa/pkg/config"
+	"github.com/openshift/rosa/pkg/fedramp"
+	"github.com/openshift/rosa/pkg/info"
+	"github.com/openshift/rosa/pkg/logging"
+	"github.com/openshift/rosa/pkg/reporter"
+)
+
+//go:generate mockgen -source=client.go -package=ocm -destination=mock_ocmclient.go
+type Client struct {
+	ocm *sdk.Connection
+}
+
+// ClientBuilder contains the information and logic needed to build a connection to OCM. Don't
+// create instances of this type directly; use the NewClient function instead.
+type ClientBuilder struct {
+	logger *logrus.Logger
+	cfg    *config.Config
+}
+
+// NewClient creates a builder that can then be used to configure and build an OCM connection.
+func NewClient() *ClientBuilder {
+	return &ClientBuilder{}
+}
+
+// NewClientWithConnection creates a client with a preexisting connection for testing purpose
+func NewClientWithConnection(connection *sdk.Connection) *Client {
+	return &Client{
+		ocm: connection,
+	}
+}
+
+func CreateNewClientOrExit(logger *logrus.Logger, reporter *reporter.Object) *Client {
+	client, err := NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+
+	return client
+}
+
+// Logger sets the logger that the connection will use to send messages to the log. This is
+// mandatory.
+func (b *ClientBuilder) Logger(value *logrus.Logger) *ClientBuilder {
+	b.logger = value
+	return b
+}
+
+// Config sets the configuration that the connection will use to authenticate the user
+func (b *ClientBuilder) Config(value *config.Config) *ClientBuilder {
+	b.cfg = value
+	return b
+}
+
+// Build uses the information stored in the builder to create a new OCM connection.
+func (b *ClientBuilder) Build() (result *Client, err error) {
+	if b.cfg == nil {
+		// Load the configuration file:
+		b.cfg, err = config.Load()
+		if err != nil {
+			err = fmt.Errorf("Failed to load config file: %v", err)
+			return nil, err
+		}
+		if b.cfg == nil {
+			err = fmt.Errorf("Not logged in, run the 'rosa login' command")
+			return nil, err
+		}
+	}
+
+	// Enable the FedRAMP flag globally
+	if b.cfg.FedRAMP {
+		fedramp.Enable()
+	}
+
+	// Check parameters:
+	if b.logger == nil {
+		err = fmt.Errorf("Logger is mandatory")
+		return
+	}
+
+	// Create the OCM logger that uses the logging framework of the project:
+	logger, err := logging.NewOCMLogger().
+		Logger(b.logger).
+		Build()
+	if err != nil {
+		return
+	}
+
+	// Prepare the builder for the connection adding only the properties that have explicit
+	// values in the configuration, so that default values won't be overridden:
+	builder := sdk.NewConnectionBuilder()
+	builder.Logger(logger)
+	builder.Agent(info.UserAgent + "/" + info.Version + " " + sdk.DefaultAgent)
```

Comment:
The way this is setup from here it is picking up rosa information instead of rhcs